ddl/partition.go

```diff
@@ -1202,6 +1213,24 @@ func onTruncateTablePartition(d *ddlCtx, t *meta.Meta, job *model.Job) (int64, e
 				}
 			}
 		}
+		tikvStore, ok := d.store.(helper.Storage)
```

Comment:
I think there is another strategy that we can firstly SetPlacementRule before modifying `tblInfo.TiFlashReplica`, and `return model.JobStateCancelled` if SetPlacementRule fails, but I am not sure whether it is safe here.